ui-admin/src/study/participants/enrolleeView/EnrolleeProfile.tsx

```diff
@@ -332,3 +300,141 @@ export function FormRow(
+      setAddressValidationResults(results)
+      setHasChangedSinceValidation([])
+    }, { setIsLoading: setIsLoadingValidation })
+  }
+
+  const formatClassName = (field: keyof MailingAddress) => {
+    if (!hasChangedSinceValidation.includes(field)
+      && addressValidationResults?.valid === true
+      && isNil(addressValidationResults.suggestedAddress)) {
+      return 'form-control is-valid'
+    }
+
+    if (!hasChangedSinceValidation.includes(field)
+      && !isAddressFieldValid(addressValidationResults, field, mailingAddress[field])) {
+      return 'form-control is-invalid'
+    }
+
+    return 'form-control'
+  }
+
+  const clearSuggestedAddress = () => {
+    setAddressValidationResults(val => {
+      return {
+        ...(val || {}), // technically it could be undefined but should be impossible when this is called
+        valid: true,
+        suggestedAddress: undefined
+      }
+    })
+  }
+
+
+  return <FormRow title={title}>
+    <div className="">
+      <div className='row mb-2'>
+        <div className="col">
+          <input
+            className={formatClassName('street1')}
+            type="text" value={mailingAddress.street1 || ''} placeholder={'Street 1'}
+            onChange={e => onFieldChange('street1', e.target.value)}/>
+        </div>
+      </div>
+      <div className='row mb-2'>
+        <div className="col">
+          <input
+            className={formatClassName('street2')}
+            type="text" value={mailingAddress.street2 || ''}
+            placeholder={'Street 2'}
+            onChange={e => onFieldChange('street2', e.target.value)}/>
+        </div>
+      </div>
+      <div className='row mb-2'>
+        <div className="col">
+          <input
+            className={formatClassName('city')}
+            type="text" value={mailingAddress.city || ''}
+            placeholder={'City'}
+            onChange={e => onFieldChange('city', e.target.value)}/>
+        </div>
+        <div className='col'>
+          <input
+            className={formatClassName('state')}
+            type="text" value={mailingAddress.state || ''} placeholder={'State/Province'}
+            onChange={e => onFieldChange('state', e.target.value)}/>
+        </div>
+      </div>
+      <div className='row'>
+        <div className="col">
+          <input
+            className={formatClassName('postalCode')}
+            type="text" value={mailingAddress.postalCode || ''} placeholder={'Postal Code'}
+            onChange={e => onFieldChange('postalCode', e.target.value)}/>
+        </div>
+        <div className='col'>
+          <input
+            className={formatClassName('country')}
+            type="text" value={mailingAddress.country || ''} placeholder={'Country'}
+            onChange={e => onFieldChange('country', e.target.value)}/>
+        </div>
+      </div>
+      <LoadingSpinner isLoading={isLoadingValidation}>
+        <button className="btn btn-link" onClick={validateAddress}>Validate</button>
```

Comment:
I wasn't sure how to make this button only show up in dev/local. Is there any easy way to do that?

Reply:
ah, yeah, this is where we need a feature flag.   We've talked about putting in feature flag infrastructure before, and it wasn't worth it back then given the already-outrageously large amount of configuration.  But this is a good case for it, we might want to think about it

In the meantime, you can just restrict this feature to superusers with `  const { user } = useUser()` and then checking user.superuser.